### PR TITLE
gui: Implement privacy mode

### LINKED ISSUE #1550: "Hidden" mode for wallet
Came across this while cleaning Gridcoin-tasks. It's a mode that can be turned on in the wallet settings where all information is asterisked out when engaged. I see it is some other wallets for other currencies. It's intended to hide sensitive information (balance, CPID, etc) from anyone standing behind you. Seems doable and a decent feature, but not very high priority.

Old Issue:
https://github.com/gridcoin-community/Gridcoin-Tasks/issues/61

### PR DESCRIPTION
**Pull request by jamescowens**
This PR is based on https://github.com/bitcoin/bitcoin/pull/16432 and implements value masking (privacy mode) for the Gridcoin overview screen.

While the PR generally follows the Bitcoin approach, the reviewers that made the comments on the Bitcoin PR that the state of privacy should be stored in the optionsModel are correct. On restart, the GUI should remember the privacy setting to avoid actually revealing balances to the people around the workstation if a wallet restart is necessary.

Note that in addition to the state maintenance in the optionsModel, I wanted to retain the hotkey approach that Bitcoin adopted. This necessitated connecting the Settings -> Mask values entry, which has a hotkey of [Shift][Ctrl][M] to the optionsModel state.

When the hotkey sequenced is pressed, the Settings->Mask values toggle is selected or the Settings->Options->Display->Mask values in the Overview screen checkbox is checked, the overview screen is masked to look like this:

image
 
The GRC values and the coin weight, magnitude, and pending reward fields are masked to #.## or # and the recent transactions are hidden.

Closes #1550.

### CODE REVIEW

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -758,6 +765,13 @@ void BitcoinGUI::setClientModel(ClientModel *clientModel)
         // Report errors from network/worker thread
         connect(clientModel, &ClientModel::error, this, &BitcoinGUI::error);
 
+        // Ensure the checkbox for mask values action matches the retrieved state from the optionsModel.
+        m_mask_values_action->setChecked(isPrivacyModeActivated());
+
+        // Connect the action to the setPrivacy function. (This has to be done after the setting of the
```
Comment (barton2526) [inline src/qt/bitcoingui.cpp:771]: hanging `(`

**Comment by div72** [comment]
@jamescowens Magnitude should also be hidden in my opinion. Also the transactions tab seems to be broken:
Screen Shot 2021-11-30 at 20 50 49

**Comment by jamescowens** [comment]
The transactions area on the main screen is supposed to be masked when privacy mode is engaged. This is intended behavior. I think I agree on magnitude.

**Comment by div72** [comment]
> The transactions area on the main screen is supposed to be masked when privacy mode is engaged. This is intended behavior.

To clarify, I am talking about the "Recent Transactions" text being in the middle, not transactions not being visible. I assume that's not intended?

**Comment by jamescowens** [comment]
Well... not exactly. The issue (I suppose if you call it that) is that the recentTransactionsFrame in overviewpage.ui currently contains four widgets at wallet start: horizontal_layout_3 (which contains the Recent Transactions header you mentioned above), recentTransactionsHeaderLine, recentTransactionsNoResult, and listTransactions.

@cyrossignol, to save memory, changed the code to actually DELETE the recentTransactionsNoResult widget once the listTransactions is non-empty. This made sense before, because the rationale is that once transactions are there, they are there.

When in combination with the privacy mode, where we are hiding listTransactions, the only widgets left are the header and the line, which will be centered. I think to fix this (and also use a different descriptive language) we should also put a fifth widget in there recentTransactionsMasked, which would be hidden with opposite polarity to listTransactions. The recentTransactionsMasked can have appropriate verbiage to explain that the list is actually masked.

**Comment by jamescowens** [comment]
Ok. I think that does it.

File: src/qt/overviewpage.cpp [hunk 2]
```diff
@@ -337,6 +384,12 @@ void OverviewPage::setWalletModel(WalletModel *model)
 
         connect(model->getOptionsModel(), &OptionsModel::LimitTxnDisplayChanged, this, &OverviewPage::updateTransactions);
         connect(model, &WalletModel::transactionUpdated, this, &OverviewPage::updateTransactions);
+
+        // Set the privacy state for the overview screen from the optionsModel for init.
+        setPrivacy(model->getOptionsModel()->getMaskValues());
+
+        // Connect the privacy mode setting to the options dialog.
+        connect(walletModel->getOptionsModel(), &OptionsModel::MaskValuesChanged, this, & OverviewPage::setPrivacy);
```
Comment (div72) [inline src/qt/overviewpage.cpp]:
```suggestion
        connect(walletModel->getOptionsModel(), &OptionsModel::MaskValuesChanged, this, &OverviewPage::setPrivacy);
```
nit: format

File: src/qt/bitcoingui.cpp (hunk 1 above)
Comment (div72) [reply src/qt/bitcoingui.cpp:771]: Still hanging.

**Review by div72 — APPROVED** [review]
ACK e952ab880e9c27e2feb68fc26c39174b90801fd7.

**Pull request merged**